create_table.lua

```diff
@@ -1,21 +1,23 @@
 box.cfg{listen = 3301, memtx_memory = 10 * 1024^3}
 box.schema.user.grant('guest', 'read,write,execute', 'universe')
 
-box.sql.execute("drop table if exists warehouse;")
-box.sql.execute("create table warehouse ( \
+local sql_execute = box.sql and box.sql.execute or box.execute
+
+sql_execute("drop table if exists warehouse;")
+sql_execute("create table warehouse ( \
 w_id int not null, \
 w_name varchar(10), \
 w_street_1 varchar(20), \
 w_street_2 varchar(20), \
 w_city varchar(20), \
 w_state varchar(2), \
 w_zip varchar(9), \
-w_tax varchar(7), \
-w_ytd varchar(15), \
+w_tax number, \
+w_ytd number, \
```

Comment:
I looked at the [revision 5.11](http://tpc.org/tpc_documents_current_versions/pdf/tpc-c_v5.11.0.pdf) of the standard, page 12:

> * `numeric(m [,n])` means an unsigned numeric value with at least `m` total decimal digits, of which `n` digits are to the right (after) the decimal point. The attribute must be able to hold all possible values which can be expressed as `numeric(m, n)`. Omitting `n`, as in `numeric(m)`, indicates the same as `numeric(m, 0)`. Numeric fields that contain monetary values (W_YTD, D_YTD, C_CREDIT_LIM, C_BALANCE, C_YTD_PAYMENT, H_AMOUNT, OL_AMOUNT, I_PRICE) must use data types that are defined by the DBMS as **being an exact numeric data type** or that satisfy the ANSI SQL Standard definition of being an exact numeric representation.
> * `signed numeric(m [,n])` is identical to `numeric(m [,n])` except that it can represent both positive and negative values.

> WAREHOUSE Table Layout
>
> | Field Name | Field Definition | Comments |
> | --- | --- | --- |
> | <...> | <...> | <...> |
> | W_TAX | signed numeric(4,4) | Sales tax |
> | W_YTD | signed numeric(12,2) | Year to date balance |

A 64-bit number holds 53 significant bits, so it should be enough to hold 14 decimal digits (2^53 > 10^14). However the cited paragraph of the TPC-C standard obligates a database to return exact representation of a number we store in it. So at least some extra client rounding is necessary to interpret the returned values as, say, `signed numeric(12,2)`:

```
$ cc -Wall -Wextra -x c <(echo -e '#include <stdio.h>\nint main() { printf("%.16lf\\n", 9999999999.99); return 0; }') && ./a.out; rm a.out
9999999999.9899997711181641
```

I didn't look how the benchmark works with the data, so maybe the result is correct and I'm wrong. But now it is not obvious for me, whether using of `number` is correct here.

Reply:
Please, also verify other schema changes for correctness regarding the standard.